File: benfica-frontend/frontend/src/app/cliente/shared/entidade/cliente.ts
import { Endereco } from "src/app/usuario/shared/entidade/endereco";
import { Seguro } from "./seguro";
import { ContatoCliente } from "./contatoCliente";

export class Cliente {

  constructor(){
    this.id = 0;
    this.nome = '';
    this.cpf = '';
    this.endereco = new Endereco();
    this.seguro = [];
    this.contatoCliente = [];
    this.tipoSeguro = 0;
  }

  id: number;
  nome: string;
  cpf: string;
  seguro: Seguro[];
  contatoCliente: ContatoCliente[];
  endereco: Endereco;
  tipoSeguro: number;
}

File: backend/src/main/java/com/benfica/model/Cliente.java
package com.benfica.model;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import org.springframework.lang.NonNull;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Cliente {


	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column
	@NonNull
	private String nome;
	
	@Column
	private String cpf;
	
	@OneToMany(mappedBy = "cliente")
	private List<ContatoCliente> contatoCliente;
	
	@OneToMany(mappedBy = "cliente", cascade = CascadeType.ALL)
	private List<Seguro> seguro;
	
	@OneToOne(cascade = CascadeType.REMOVE)
	private Endereco endereco;
}
